### DISCUSSION TITLE
List of checks enabled by default?

### DISCUSSION THREAD

**Discussion by JoeStainesAtONT** [Ericsson/codechecker#3085] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I may be missing it from the documentation but is there any description or any indication of what checks are enabled by default without any configuration file?

**Describe the solution you would like**
Somewhere in the documentation describing the default checks that are enabled for clang tidy

**Describe alternatives you have considered**
-

**Additional context**
-

**Comment by csordasmarton** [comment, marked as answer]
@JoeStainesAtONT If you run the following command it will show you the list of checkers for Clang Tidy: `CodeChecker checkers --analyzer clang-tidy --details`.

If a line starts with `+` it means it is enabled by default.

**Comment by JoeStainesAtONT** [comment]
That's perfect, thanks @csordasmarton. As i imagine the defaults can change between releases possibly, is it possible to get that line added to the documentation (perhaps within https://github.com/Ericsson/codechecker/blob/master/docs/analyzer/checker_and_analyzer_configuration.md) so others will be able to check easily?

**Comment by whisperity** [comment]
Clang-Tidy does **not** supply a list of "default enabled checkers". As far as I know, in Clang-Tidy's case, either nothing or everything is enabled by default. The list of checkers that CodeChecker reports to be enabled by default in Clang-Tidy is **our** gathering! There always *is* a configuration file, supplied with CodeChecker itself, that sets these profiles:

https://github.com/Ericsson/codechecker/blob/c374e323c856aeccb647dfe6f2efd432f2122cf8/config/checker_profile_map.json#L100-L101
